src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs

```diff
@@ -123,7 +123,7 @@ public void Initialize()
             _applicationLifetime?.NotifyStarted();
 
             // Fire IHostedService.Start
-            _hostedServiceExecutor.Start();
+            await _hostedServiceExecutor.StartAsync(cancellationToken);
```

Comment:
Ya everything ok the start and stop path needs this because of the xunit sync context.